internal/dao/pgx/reservation_pgx.go

```diff
@@ -27,23 +27,31 @@ func getReservationDao(ctx context.Context) dao.ReservationDao {
 }
 
 func (x *reservationDao) CreateNoop(ctx context.Context, reservation *models.NoopReservation) error {
-	reservation.Provider = models.ProviderTypeNoop
-	if err := x.createGenericReservation(ctx, &reservation.Reservation); err != nil {
-		return fmt.Errorf("failed to create reservation record: %w", err)
+	txErr := dao.WithTransaction(ctx, func(tx pgx.Tx) error {
+		reservation.Provider = models.ProviderTypeNoop
+		if err := x.createGenericReservation(ctx, tx, &reservation.Reservation); err != nil {
+			return fmt.Errorf("failed to create reservation record: %w", err)
+		}
+
+		return nil
+	})
+
+	if txErr != nil {
+		return fmt.Errorf("pgx tx error: %w", txErr)
 	}
 	return nil
```

Comment:
To make things simple and because pgx.Tx and pgx.Pool are incompatible types, I am putting this in tx block although it does not need to be. I think I could have used pgx.Querier or some interface, not sure.

Reply:
Nah Querier is not from the pgx package, I would rather not use it. These are two different types.